python-mini-projects/projects/Store_emails_in_csv/store_emails.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+
+import csv
+import email
+from email import policy
+import imaplib
+import logging
+import os
+import ssl
+
+from bs4 import BeautifulSoup
+
+
+credential_path = "credentials.txt"
+csv_path = "mails.csv"
+
+logger = logging.getLogger('imap_poller')
+
+host = "imap.gmail.com"
+port = 993
+ssl_context = ssl.create_default_context()
+
+
+def connect_to_mailbox():
+    # get mail connection
+    mail = imaplib.IMAP4_SSL(host, port, ssl_context=ssl_context)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *mail*. The resource is allocated by call *imaplib.IMAP4_SSL*. Currently, there are execution paths that do not contain closure statements, for example, when *IMAP4\_SSL.login()* throws an exception. To prevent this resource leak, close *mail* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).